lib/ui/pages/news_list/news_list_page.dart

```diff
@@ -25,54 +25,61 @@ class _NewsPageState extends State<NewsListPage> {
         child: Column(
           crossAxisAlignment: CrossAxisAlignment.start,
           children: [
-            PrimaryNewsListItem(
-              news: PrimaryNewsListEntity(
-                imageUrl: 'https://i.picsum.photos/id/9/800/800.jpg',
-                title: "New laptops seem to be stuck to the board",
-                authorName: "Marek Aureliusz",
-                authorImageUrl:
-                    'https://www.ludoviccareme.com/files/image_211_image_fr.jpg',
-                url:
-                    'https://www.nytimes.com/2020/05/14/obituaries/jonathan-adewumi-dies-coronavirus.html',
-              ),
+            Container(
+              width: double.maxFinite,
+              height: 80,
+            ),
+            _buildNewsSectionHeader(
+              title: Strings.mostViewed,
+              route: null,
+            ),
+            SizedBox(height: _Constants.sectionHeaderPadding),
+            PrimaryNewsListView(
+              primaryNewsListEntities: primaryNewsListEntities.take(5).toList(),
+            ),
+            _buildNewsSectionHeader(
+              title: Strings.newest,
+              route: (context) => SecondaryNewsListPage(),
             ),
-            _buildSecondaryNewsSectionHeader(),
             SizedBox(height: _Constants.sectionHeaderPadding),
             SecondaryNewsListView(
-              secondaryNewsListEntities:
-                  secondaryNewsListEntities.take(3).toList(),
+              secondaryNewsListEntities: secondaryNewsListEntities.take(3).toList(),
             ),
           ],
         ),
       ),
     );
   }
 
-  Widget _buildSecondaryNewsSectionHeader() => Padding(
-        padding: EdgeInsets.symmetric(horizontal: Dimens.pagePadding),
-        child: Row(
-          mainAxisAlignment: MainAxisAlignment.spaceBetween,
-          crossAxisAlignment: CrossAxisAlignment.center,
-          children: [
-            Hero(
-              tag: Strings.newest,
-              child: Text(
-                Strings.newest,
-                style: Theme.of(context).textTheme.headline3,
-              ),
+  Widget _buildNewsSectionHeader({
+    @required String title,
+    @required Function route,
```

Comment:
wydaje mi się, że jak funkcja ma coś przyjmować/zwracać to warto określić jawnie sygnaturę czyli w tym przypadku `Widget Function(BuildContext)`, jest na to typedef WidgetBuilder więc zamiast Function wskazałbym jawnie typ argumentu jako WidgetBuilder :) druga rzecz to może zmieniłbym nazwę bo route niekoniecznie brzmi jak funkcja, może routeBuilder?